### DISCUSSION TITLE
Audio Pin Definition

### DISCUSSION THREAD

**Discussion by DanMan32** [PlummersSoftwareLLC/NightDriverStrip#203] [Q&A]
Where in NightDriver is the GPIO for audio defined?
I didn't find it in PlatformIO.ini or globals.h

**Comment by davepl** [comment]
INPUT_PIN

**Reply by stretch911** [reply]
probably in main...but here's the magic of a real ide (like vs code)....find the item you want to "investigate"...hover over it...if its a constant (as if you follow Dave's code discipline https://learn.microsoft.com/en-us/windows/win32/stg/coding-style-conventions?redirectedfrom=MSDN you can tell by the NAME its all capped...probably a constant. so when you hover over it, it should show you the value...
if its a function/method/class...right click on it, and on the drop down menu click goto definition. this will take you to where ever the code is defined....pretty cool. 

edit: also pay attention to color coding...like things are colored the same...

**Comment by shin-tinh-nghich** [comment]
input_pin and [adc1_channel_0](https://github.com/PlummersSoftwareLLC/NightDriverStrip/blob/08d1895783da955584534930ee3e329b6b87ec2e/include/soundanalyzer.h#L587)
